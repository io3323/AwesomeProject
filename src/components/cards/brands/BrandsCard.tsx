import React, {FC} from 'react';
import {Text, TouchableOpacity} from 'react-native';
import {styles} from './styles';
import {BrandsType} from './types';

export const BrandsCard: FC<BrandsType> = ({
  brandId,
  brandName,
  selectBrand,
  setSelectBrand,
}) => {
  return (
    <TouchableOpacity
      onPress={() =>
        setSelectBrand(
          brandId === selectBrand.brandId
            ? {}
            : {brandId: brandId, brandName: brandName},
        )
      }
      style={
        brandId === selectBrand.brandId
          ? styles.activeCardContainer
          : styles.cardContainer
      }>
      <Text>{brandName}</Text>
    </TouchableOpacity>
  );
};
