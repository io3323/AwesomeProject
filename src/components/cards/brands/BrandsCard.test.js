import React from 'react';
import renderer from 'react-test-renderer';
import {BrandsCard} from './BrandsCard';

test('BrandsCard Component', () => {
  // Mock props
  const brandId = 1;
  const brandName = 'BrandName';
  const selectBrand = {brandId: 1, brandName: 'SelectedBrand'};
  const setSelectBrand = jest.fn();

  // Render the component
  const component = renderer.create(
    <BrandsCard
      brandId={brandId}
      brandName={brandName}
      selectBrand={selectBrand}
      setSelectBrand={setSelectBrand}
    />,
  );

  // Get the JSON representation of the component
  const tree = component.toJSON();

  // Expect the component to match the snapshot
  expect(tree).toMatchSnapshot();

  // Simulate press
  tree.props.onPress();

  // Ensure that setSelectBrand was called with the correct parameters
  expect(setSelectBrand).toHaveBeenCalledWith(
    brandId === selectBrand.brandId ? {} : {brandId, brandName},
  );
});
