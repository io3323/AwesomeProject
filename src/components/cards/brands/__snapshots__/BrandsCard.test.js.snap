// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandsCard Component 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderColor": "blue",
      "borderRadius": 3,
      "borderStyle": "solid",
      "borderWidth": 1,
      "display": "flex",
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 30,
      "paddingVertical": 10,
    }
  }
>
  <Text>
    BrandName
  </Text>
</View>
`;

exports[`BrandsCard Component handles press correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderColor": "black",
      "borderRadius": 3,
      "borderStyle": "solid",
      "borderWidth": 1,
      "display": "flex",
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 30,
      "paddingVertical": 10,
    }
  }
>
  <Text>
    TestBrand
  </Text>
</View>
`;

exports[`BrandsCard Component renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderColor": "blue",
      "borderRadius": 3,
      "borderStyle": "solid",
      "borderWidth": 1,
      "display": "flex",
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 30,
      "paddingVertical": 10,
    }
  }
>
  <Text>
    TestBrand
  </Text>
</View>
`;
